clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -39,57 +38,104 @@
     private CompressionType compression;
     private byte magic;
     private long firstOffset;
+    private long pid;
+    private short epoch;
+    private int firstSequence;
+    private long logAppendTime = System.currentTimeMillis();
+    private int partitionLeaderEpoch = 998;
 
     public MemoryRecordsTest(byte magic, long firstOffset, CompressionType compression) {
         this.magic = magic;
         this.compression = compression;
         this.firstOffset = firstOffset;
+        if (magic >= RecordBatch.MAGIC_VALUE_V2) {
+            pid = 134234L;
+            epoch = 28;
+            firstSequence = 777;
+        } else {
+            pid = RecordBatch.NO_PRODUCER_ID;
+            epoch = RecordBatch.NO_PRODUCER_EPOCH;
+            firstSequence = RecordBatch.NO_SEQUENCE;
+        }
     }
 
     @Test
     public void testIterator() {
-        MemoryRecordsBuilder builder1 = MemoryRecords.builder(ByteBuffer.allocate(1024), magic, compression, TimestampType.CREATE_TIME, firstOffset);
-        MemoryRecordsBuilder builder2 = MemoryRecords.builder(ByteBuffer.allocate(1024), magic, compression, TimestampType.CREATE_TIME, firstOffset);
```

Comment:
I think we were verifying the auto-incrementing behavior. I will verify that we have both cases covered.